### ISSUE TITLE
FIPS enable computers fail due to md5 use

### ISSUE THREAD

**Issue by g19fanatic** [tensorflow/tensorflow#31800]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Red Hat Enterprise Linux Workstation release 7.7
```
Python 2.7.5 (default, Jun 11 2019, 14:33:56)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-39)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow as tf
>>> tf.version.GIT_VERSION
'v1.14.0-rc1-22-gaf24dc91b5'
```
- Installed using virtualenv? pip? conda?: pipenv install tensorflow

tflearn fails to import due to tensorflows use of md5 on an fips enabled machine
```
Python 2.7.5 (default, Jun 11 2019, 14:33:56)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-39)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import tflearn
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../lib/python2.7/site-packages/tflearn/__init__.py", line 4, in <module>
    from . import config
  File ".../lib/python2.7/site-packages/tflearn/config.py", line 5, in <module>
    from .variables import variable
  File ".../lib/python2.7/site-packages/tflearn/variables.py", line 7, in <module>
    from tensorflow.contrib.framework.python.ops import add_arg_scope as contrib_add_arg_scope
  File ".../lib/python2.7/site-packages/tensorflow/contrib/__init__.py", line 31, in <module>
    from tensorflow.contrib import cloud
  File ".../lib/python2.7/site-packages/tensorflow/contrib/cloud/__init__.py", line 28, in <module>
    from tensorflow.contrib.bigtable.python.ops.bigtable_api import BigtableClient
  File ".../lib/python2.7/site-packages/tensorflow/contrib/bigtable/__init__.py", line 29, in <module>
    from tensorflow.contrib.bigtable.python.ops.bigtable_api import BigtableClient
  File ".../lib/python2.7/site-packages/tensorflow/contrib/bigtable/python/ops/bigtable_api.py", line 44, in <module>
    resource_loader.get_path_to_datafile("_bigtable.so"))
  File ".../lib/python2.7/site-packages/tensorflow/contrib/util/loader.py", line 56, in load_op_library
    ret = load_library.load_op_library(path)
  File ".../lib/python2.7/site-packages/tensorflow/python/framework/load_library.py", line 73, in load_op_library
    module_name = hashlib.md5(wrappers).hexdigest()
ValueError: error:060800A3:digital envelope routines:EVP_DigestInit_ex:disabled for fips
```
replacing all md5 calls with sha1 calls should work?

**Comment by ajschumacher** [comment]
Thanks for creating the issue, @g19fanatic! Thanks for looking into it, @mihaimaruseac! Recently encountered this as well. Currently manually editing the source code to achieve FIPS compliance, but it would be far better for TensorFlow to work with FIPS out of the box.

**Comment by rusu24edward** [comment]
I ran into this in load_library. Adding the optional argument `used_for_security=False` solved this for me.

**Comment by demongolem** [comment]
Any movement on this?  I still get it importing the latest version of TF.

**Comment by mihaimaruseac** [comment]
#31803 was an attempt at this but it required more work due to internal usages.

I'm currently working on that, should be done by the time TF 2.5 lands

**Comment by g19fanatic** [comment]
I am confused as to what was done to fix this and why my issue was closed?
https://github.com/tensorflow/tensorflow/pull/31803/files
So instead of merging in my PR, you do the same thing (but in less places?).
Why couldn't this have been merged over a year ago if you were just going to do the same thing?

**Comment by mihaimaruseac** [comment]
There are internal fixes in parallel with these ones that didn't get exposed.

Also, a year ago there were more files that needed change but they got replaced in the meanwhile.

There is one more public change to arrive, was delayed by a broken test.

**Comment by sxnguyen** [comment]
Hi,
Can someone confirm that TF 2.5 has this fix? Reading the release note for 2.5, and I don't see FIPS mentioned.
Thanks.

**Comment by jprosser** [comment]
I can confirm that with v2.5 I no longer see the FIPS errors whereas with 2.4.1 I do.

2.4.1 errors:
File "/data/users/jprosser/tf/venv/lib64/python3.6/site-packages/tensorflow/python/framework/load_library.py", line 67, in load_op_library
    module_name = hashlib.md5(wrappers).hexdigest()
ValueError: error:060800A3:digital envelope routines:EVP_DigestInit_ex:disabled for fips
